src/ResourceManager/Compute/Commands.Compute/Models/AzureDiskEncryptionStatusContext.cs

```diff
@@ -3,16 +3,24 @@
 
 namespace Microsoft.Azure.Commands.Compute.Models
 {
+    enum EncryptionStatus
+    {
+        Encrypted,
+        NotEncrypted,
+        Unknown
+    }
+
     class AzureDiskEncryptionStatusContext
     {
-        public bool OsVolumeEncrypted { get; set; }
+        public EncryptionStatus OsVolumeEncrypted { get; set; }
         public DiskEncryptionSettings OsVolumeEncryptionSettings { get; set; }
-
-        [JsonIgnore]
-        public string OsVolumeEncryptionSettingsText
-        {
-            get { return JsonConvert.SerializeObject(OsVolumeEncryptionSettings, Formatting.Indented); }
-        }
-        public bool DataVolumesEncrypted { get; set; }
+        public EncryptionStatus DataVolumesEncrypted { get; set; }
+    }
+    class AzureDiskEncryptionStatusLinuxContext
+    {
```

Comment:
Do you really need a separate Context class for Linux?  I think you can use existing AzureDiskEncryptionStatusContext for Linux, by adding DataVolumeEncryptionSettings property.
For Windows, just set DataVolumeEncryptionSettings to null.